### PR TITLE
SINGA-433 updated documentation about docker, dockerfile for consitent, cudnn version in dockerfile

### PR DESCRIPTION
**Pull request by dcslin**
Hi team, this is to update docker doc as well as dockerfile arrangement to make them consistent

### CODE REVIEW

**Comment by dcslin** [comment]
tested dockerfiles, `cuda9, cuda10, mkldnn` and error free in unit test
example:
```
docker build --no-cache `pwd`/incubator-singa/tool/docker/devel/ubuntu/cuda9 -t singa:devel-cuda9-cudnn
docker run -it --runtime nvidia --name singa_cuda9 --rm singa:devel-cuda9-cudnn7    /home/singa/incubator-singa/build/bin/test_singa
```

File: doc/en/docs/docker.md [hunk 1]
```diff
@@ -31,28 +31,25 @@ Dockerfile folder, e.g., tool/docker/devel/
...
+* CUDNN: cudnn7
+* MKLDNN: mkldnn0.18
 
-By default, if the version is not included in the tag, the latest stable version is used.
-The default OS is Ubuntu. The version is the latest stable version (e.g., 16.04 for now).
-For -cuda version, the **cudnn** is included by default. Their versions are also the latest stable version, i.e., cuda8.0 and cudnn5 for now.
+Here are some example tags:
 
-Here are some example tags,
+`devel-cuda9-cudnn7`, `devel-cuda9-cudnn7`, `devel-cuda10-cudnn7`, `devel-mkldnn0.18`, `devel-centos6`, `runtime`
```
Comment (nudles) [inline doc/en/docs/docker.md]: I think we can just provide devel-cuda9-cudnn7, devel-cuda10-cudnn7, devel-cpu, runtime-gpu (use cuda10-cudnn7, conda install) and runtime-cpu (conda install),.
mkldnn is enabled by default.

Comment (dcslin) [reply doc/en/docs/docker.md]: Hi @nudles Thank you for the review. This is updated.

**Comment by nudles** [comment]
1. for the cpu runtime, pls add the instructions to install singa via `conda install -c nusdbsystem singa-cpu`
2. for the gpu runtime, pls add the instructions to install singa via `conda install -c nusdbsystem singa-gpu`
3. update the tag name to be apache/singa:xxx https://issues.apache.org/jira/projects/SINGA/issues/SINGA-428?filter=allopenissues

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -144,9 +144,8 @@ ENDIF()
 
 IF(USE_MKLDNN)
     FIND_PATH(MKLDNN_INCLUDE_DIR NAME "mkldnn.hpp" PATHS "$ENV{CMAKE_INCLUDE_PATH}")
-    FIND_LIBRARY(MKLDNN_LIBRARIES NAME "libmkldnn.so" PATHS "$ENV{CMAKE_LIBRARY_PATH}")
-    FIND_LIBRARY(MKLML_LIBRARIES NAME "libmklml_intel.so" PATHS "$ENV{CMAKE_LIBRARY_PATH}")
```
Comment (nudles) [inline cmake/Dependencies.cmake]: if you remote libmklml_intel.so, can you pass the c++ test?

Comment (dcslin) [reply cmake/Dependencies.cmake]: Hi @nudles , thank you for the review. This is done in my local. I was testing OSX build process. I will fix this ticket soon.

Comment (nudles) [reply cmake/Dependencies.cmake]: I meant the test in ubuntu docker. I will do the test on macos.

Comment (dcslin) [reply cmake/Dependencies.cmake]: Yes. the test works without MLKML. I have removed MKLML in cmake and dockerfile. All 5 dockerfiles are tested and testing are passing. 
The last thing left is testing on OSX for now.

**Pull request merged**